CHANGES.md

```diff
@@ -12,21 +12,107 @@ To be released.
+       Address?, bool)` constructor.
+ -  Constructors of `Currency` class now take `(BigInteger Major),
+    BigInteger Minor)?` as the maximum supply parameter, and a boolean value to
+    mark whether the `Currency` is trackable.  [[#915], [#2200]]
+    -  *NOTE:* if you already have defined some `Currency` instances in prior
+       to the addition of maximum supply, and if it is on a live chain, you
+       cannot modify the already-defined `Currency` instances to have maximum
+       supply caps since it would make those `Currency` instances to be
+       effectively different ones compared to the original ones, as it would
+       change their `Hash`es. You must pass a `null` as their maximum supply
+       parameter. Also, you must set the `totalSupplyTrackable` parameter to
+       false.
+    -  Removed `Currency(string, byte, IImutableSet<Address>?)` constructor.
+    -  Removed `Currency(string, byte, Address?)` constructor.
+    -  Added `Currency(string, byte, (BigInteger Major, BigInteger Minor)?,
+       IImutableSet<Address>?, bool)` constructor.
+    -  Added `Currency(string, byte, (BigInteger Major, BigInteger Minor)?,
+       Address?, bool)` constructor.
+ -  Added `IAccountStateDelta.TotalSupplyUpdatedCurrencies` property.
+    [[#915], [#2200]]
+ -  Added `IAccountStateView.GetTotalSupply(Currency)` method.
+    [[#915], [#2200]]
+ -  Added `IBlockChainStates<T>.GetTotalSupply(Currency, BlockHash,
+    TotalSupplyStateCompleter<T>` method which gets the total supply of
+    a `Currency` in `FungibleAssetValue` from the state, and if not found,
+    returns null.  [[#915], [#2200]]
+
 ### Backward-incompatible network protocol changes
 
+ -  The `Block<T>.CurrentProtocolVersion` is bumped from 3 to 4.  [[#2200]]
+    -  `IAccountStateDelta.GetTotalSupply` returns the current total supply
+       of a `Currency` if the total supply of`Currency` is trackable from
+       version 4, and throws `NotSupportedException` below version 4.
+    -  `IAccountStateDelta.MintAsset` and `IAccountStateDelta.BurnAsset` tracks
+       the total supply if trackable from version 4.
+
 ### Backward-incompatible storage format changes
 
 ### Added APIs
 
+ -  Added `Currency.MaximumSupply` property.  [[#915], [#2200]]
+ -  Added `Currency.TotalSupplyTrackable` field.  [[#915], [#2200]]
+ -  Added `SupplyOverflowException` class.  [[#915], [#2200]]
+ -  Added `TotalSupplyGetter` delegate.  [[#915], [#2200]]
+ -  Added `TotalSupplyStateCompleter<T>` delegate.  [[#915], [#2200]]
+ -  Added `TotalSupplyStateCompleters<T>` static class.  [[#915], [#2200]]
+ -  Added `StateCompleterSet<T>.TotalSupplyStateCompleter` property.
+    [[#915], [#2200]]
+
 ### Behavioral changes
 
+ -  Bencodex related methods in `Currency` now accounts for the maximum
+    supply and total supply tracking.  [[#915], [#2200]]
+    -  `Currency.Serialize()` method now stores the maximum supply as
+       non-nullable Bencodex `Integer` values under the keys
```

Comment:
This is a bit misleading. I think "non-nullable" should be removed or the surrounding context should be rephrased as Bencodex dictionary values are type agnostic.

Reply:
Would something like:

-  If an instance of `Currency` has a maximum supply value, `Currency.Serialize()` method now stores it as
    `Integer` values under the keys
    `maximumSupplyMajor` and `maximumSupplyMinor`. Otherwise, the entries are omitted.

be ok?